File: apps/backend/src/EnterpriseAgentOs.Domain/features/Management/Dtos/SessionTokenHasher.cs
using System.Security.Cryptography;
using System.Text;

namespace EnterpriseAgentOs.Domain.Features.Management;

public static class SessionTokenHasher
{
    public static string Hash(string token)
    {
        var bytes = SHA256.HashData(Encoding.UTF8.GetBytes(token));
        return Convert.ToHexStringLower(bytes);
    }
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/SessionTokenHasher.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public final class SessionTokenHasher {
    private SessionTokenHasher() {
    }

    public static String hash(String token) {
        try {
            byte[] digest = MessageDigest.getInstance("SHA-256").digest(token.getBytes(StandardCharsets.UTF_8));
            StringBuilder hex = new StringBuilder(digest.length * 2);
            for (byte b : digest) {
                hex.append(String.format("%02x", b));
            }
            return hex.toString();
        } catch (NoSuchAlgorithmException exception) {
            throw new IllegalStateException("SHA-256 is not available.", exception);
        }
    }
}
